src/Symfony/Component/Yaml/Inline.php

```diff
@@ -209,7 +209,7 @@ public static function parseScalar($scalar, $delimiters = null, $stringDelimiter
                 $i += strlen($output);
 
                 // remove comments
-                if (false !== $strpos = strpos($output, ' #')) {
+                if ((false !== $strpos = strpos($output, ' #')) || (false !== $strpos = strpos($output, "\t#"))) {
```

Comment:
the case being broke is when you have ``\t# #``, where it would use the second one. We need to check for both positions and use the smaller one.